Added a description how to speed up testing with parallel execution (#7516)
jenkins-infra/jenkins.io: content/doc/developer/testing/index.adoc

//==== From the Command Line
//==== From an IDE

=== Performance considerations

The test runner (Surefire) supports running tests in parallel to speed them up.

> A possibility to configure this machine wide is to adjust the `forkCount` within maven.

Reviewer: ```suggestion
A possibility to configure this machine-wide is to adjust the `forkCount` within Maven.
```

The default setting is `forkCount=1`, which means no parallel testing.

> An often used setting is `1C`, which spawns as many testing processes in parallel as cpu cores are present.

Reviewer: ```suggestion
An often-used setting is `1C`, which spawns as many testing processes in parallel as CPU cores are present.
```

> If you have a machine with many cores it might be faster to set it to a smaller number, like `0.45C`.

Reviewer 2: ```suggestion
If you have a machine with many cores, it might be faster to set it to a smaller number, like `0.45C`.
```
small adjustment for punctuation

> For example with 16 cores the runner will be spawning up to 7 processes.

Reviewer 2: ```suggestion
For example, with 16 cores the runner will be spawning up to 7 processes.
```
Small adjustment for punctuation

> More details can be found at https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html[maven surefire documentation].

Reviewer: ```suggestion
More details can be found at link:https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html[maven surefire documentation].
```

> More details can be found at link:https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html[maven surefire documentation].

Reviewer 2: ```suggestion
More details can be found in the link:https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html[Maven Surefire documentation].
```
Some adjustments for grammar and capitalization

> You can also adjust this globally within a maven profile setting, independently of the Plugin configuration.

Reviewer: ```suggestion
You can also adjust this globally within a Maven profile setting, independently of the Plugin configuration.
```

> The maven profile can be typically found at: `~/.m2/settings.xml` (Linux) or `%userprofile%/.m2` (Windows).

Reviewer: ```suggestion
The Maven profile can be typically found at: `~/.m2/settings.xml` (Linux) or `%userprofile%/.m2` (Windows).
```
